service/src/main/java/com/commercetools/pspadapter/payone/config/ServiceConfig.java

```diff
@@ -11,76 +16,44 @@
-    private final String ctClientSecret;
-    private final boolean startFromScratch;
+
+    private final List<String> tenants;
+
     private final String scheduledJobCronShortTimeFrame;
     private final String scheduledJobCronLongTimeFrame;
-    private final String secureKey;
-    private final boolean updateOrderPaymentState;
 
     /**
      * Initializes the configuration.
      *
      * @param propertyProvider to get the parameters from
-     *
      * @throws IllegalStateException if a mandatory parameter is undefined or empty
      */
     public ServiceConfig(final PropertyProvider propertyProvider) {
-        ctProjectKey = propertyProvider.getMandatoryNonEmptyProperty(PropertyProvider.CT_PROJECT_KEY);
-        ctClientId  = propertyProvider.getMandatoryNonEmptyProperty(PropertyProvider.CT_CLIENT_ID);
-        ctClientSecret = propertyProvider.getMandatoryNonEmptyProperty(PropertyProvider.CT_CLIENT_SECRET);
-        scheduledJobCronShortTimeFrame =
+
+        this.tenants = getMandatoryTenantNames(propertyProvider);
+
+        this.scheduledJobCronShortTimeFrame =
                 propertyProvider.getProperty(PropertyProvider.SHORT_TIME_FRAME_SCHEDULED_JOB_CRON)
                         .map(String::valueOf)
                         .orElse("0/30 * * * * ? *");
-        scheduledJobCronLongTimeFrame =
+        this.scheduledJobCronLongTimeFrame =
                 propertyProvider.getProperty(PropertyProvider.LONG_TIME_FRAME_SCHEDULED_JOB_CRON)
                         .map(String::valueOf)
                         .orElse("5 0 0/1 * * ? *");
-        startFromScratch = propertyProvider.getProperty(PropertyProvider.CT_START_FROM_SCRATCH)
-                .map(Boolean::valueOf)
-                .orElse(false);
-
-        secureKey = propertyProvider.getProperty(PropertyProvider.SECURE_KEY)
-                .map(String::valueOf)
-                .orElse("");
-
-        updateOrderPaymentState = propertyProvider.getProperty(PropertyProvider.UPDATE_ORDER_PAYMENT_STATE)
-                                .map(String::trim)
-                                .map(Boolean::valueOf)
-                                .orElse(false);
     }
 
     /**
-     * Gets the commercetools project key.
-     * @return the key of the commercetools project to connect with
+     * @return Non-empty list of tenant names, specified in {@code TENANTS} configuration property.
      */
-    public String getCtProjectKey() {
-        return ctProjectKey;
-    }
-
-    /**
-     * Gets the commercetools client ID.
-     * @return the client ID of the commercetools project to connect with
-     */
-    public String getCtClientId() {
-        return ctClientId;
-    }
-
-    /**
-     * Gets the commercetools client secret.
-     * @return the client password for the commercetools project to connect with
-     */
-    public String getCtClientSecret() {
-        return ctClientSecret;
+    public List<String> getTenants() {
+        return tenants;
```

Comment:
actually, it is kind of common pattern: you use `this` keyword in constructors and setters to avoid name collisions with the incoming argument names, but in all other places you omit it as it becomes redundant. 

I mostly keep `this` keyword in the constructors to avoid nasty typo mistakes if later someone re-factors a constructor/setter and misses the name collision. It is kind of security from accidental problems.